addons/udes_stock/models/stock_location.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+import os.path
+from odoo import fields, models
+
+
+class StockLocation(models.Model):
+    _name = "stock.location"
+    # Add messages and abstract model to locations
+    _inherit = ["stock.location", "mail.thread", "mixin.stock.model"]
+
+    # Disable translation instead of renaming.
+    name = fields.Char(translate=False)
+    # Add tracking for archiving.
+    active = fields.Boolean(tracking=True)
+
+    def get_common_ancestor(self):
+        """
+        Returns the smallest location containing all the locations in self.
+        Locations are considered to contain themselves.
+
+        :returns:
+            The stock.location containing all the locations in self,
+            or an empty stock.location recordset if there is no such location.
+        """
+        Location = self.env["stock.location"]
+
+        if len(self) <= 1:
+            return self
+
+        # Each location's parent_path is a "/"-delimited string of parent ids
+        # including itself
+        common_path = os.path.commonpath(self.mapped("parent_path"))
```

Comment:
I think from odoo13 no need to mapp if i am not wrong :)
```suggestion
        common_path = os.path.commonpath(self.parent_path)
```

Reply:
```python
os.path.commonpath(self.parent_path)
Traceback (most recent call last):
  File "/home/palabaster/projects/udes-vagrant/code/udes14/odoo/odoo/models.py", line 4986, in ensure_one
    _id, = self._ids
ValueError: too many values to unpack (expected 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/palabaster/projects/udes-vagrant/code/udes14/odoo/odoo/fields.py", line 930, in __get__
    record.ensure_one()
  File "/home/palabaster/projects/udes-vagrant/code/udes14/odoo/odoo/models.py", line 4989, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: stock.location(3702, 3703)
```
does not work as `self` is > 1

Reply:
I have been told that no need to use mapping in odoo13 and above :)

Reply:
note: discussed further in skype. Odoo 13 onwards allows not using .mapped() for relational fields (m2o, o2m) but not for normal fields like char / boolean:
```python
self
stock.location(4166, 4167)
self.location_id
stock.location(4164,)
self.location_id.child_ids
stock.location(4166, 4167)
self.name --> traceback
```